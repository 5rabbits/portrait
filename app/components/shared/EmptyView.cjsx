React = require('react')

@EmptyView = React.createClass
  render: ->
    return (
      <div className="empty-view">
        <div className="empty-container">
          <div className="empty-dialog">
            <i className="fa fa-clock-o fa-10x color-soft mb-md"></i>
            <h2 className="type-headline mb-xs">Aquí aparecerán tus trabajos para que puedas revisarlos.</h2>
            <p className="color-secondary mb-md">Ingresa tu primer trabajo y vuelve luego sagittis lacus augue laoreet rutrum faucibus dolor auctor.</p>
            <a href="#" className="btn btn-primary btn-lg">Agrega tu primer entrada de tiempo</a>
          </div>
        </div>
      </div>
    )

module.exports = @EmptyView
