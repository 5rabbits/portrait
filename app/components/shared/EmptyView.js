import React, { PureComponent } from 'react';

export default class EmptyView extends PureComponent {
  render() {
    return (
      <div className="empty-view">
        <div className="empty-container">
          <div className="empty-dialog">
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}
