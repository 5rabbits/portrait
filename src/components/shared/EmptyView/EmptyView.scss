@import '~styles/core';

.EmptyView {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.EmptyView__icon {
  font-size: 7em;
  margin-bottom: 24px;
  color: $default-disabled-color;

  svg {
    height: 1em !important;
    width: auto !important;
    fill: $default-disabled-color !important;
    display: block;
  }

  path {
    fill: inherit !important;
  }
}

.EmptyView__title {
  font-size: 28px;
  font-weight: 700;
}

.EmptyView__text {
  margin-top: 16px;
  font-size: 16px;
  color: $default-secondary-text-color;
}

.EmptyView__action {
  margin-top: 32px;
}
