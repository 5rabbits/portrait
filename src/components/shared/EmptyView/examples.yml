-
  description: With default props.
  code: >
    <div style={{ height: 400 }}>
      <EmptyView />
    </div>

-
  description: With custom props.
  code: >
    <div style={{ height: 400 }}>
      <EmptyView
        title="No projects found"
        text="Try changing the selected filters"
        icon={<i className="fa fa-filter" />}
        action={
          <button type="button" className="btn btn-primary btn-lg">
            Reset filters
          </button>
        }
      />
    </div>
