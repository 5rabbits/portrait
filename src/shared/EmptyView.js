import React from 'react'

const EmptyView = () =>
  <div className="empty-view">
    <div className="empty-container">
      <div className="empty-dialog">
        {this.props.children}
      </div>
    </div>
  </div>

export default EmptyView
