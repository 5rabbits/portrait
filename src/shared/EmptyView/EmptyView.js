import React from 'react'
import childrenPropType from 'propTypes/children'

const EmptyView = props =>
  <div className="empty-view">
    <div className="empty-container">
      <div className="empty-dialog">
        {props.children}
      </div>
    </div>
  </div>

EmptyView.propTypes = {
  children: childrenPropType,
}

export default EmptyView
