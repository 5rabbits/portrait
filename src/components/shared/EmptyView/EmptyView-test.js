import { React, toJSON, mount } from 'test/helper'
import EmptyView from 'shared/EmptyView'

describe('EmptyView', () => {
  it('should render with minimum props', () => {
    const component = mount(
      <EmptyView />,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should allow to customize all info', () => {
    const component = mount(
      <EmptyView
        icon={<i className="fa fa-search" />}
        title="Some title"
        text="Some title"
        action={<button />}
      />,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
