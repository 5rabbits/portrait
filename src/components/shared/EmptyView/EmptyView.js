import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import SearchIcon from 'icons/Search'
import cx from 'classnames'
import i18n from 'helpers/i18n'
import './EmptyView.scss'

/**
 * Used to display an empty results page on a filtered list.
 */
export default class EmptyView extends PureComponent {
  static propTypes = {
    /**
     * CSS class assigned to the container.
     */
    className: PropTypes.string,

    /**
     * Icon to be displayed.
     */
    icon: PropTypes.node,

    /**
     * Title to be displayed.
     */
    title: PropTypes.string,

    /**
     * Text to be displayed.
     */
    text: PropTypes.string,

    /**
     * Action to be displayed.
     */
    action: PropTypes.node,
  }

  static defaultProps = {
    className: null,
    icon: <SearchIcon />,
    title: null,
    text: null,
    action: null,
  }

  render() {
    const { className, icon, action, ...other } = this.props
    const title = this.props.title || i18n.t('EmptyView.title')
    const text = this.props.text || i18n.t('EmptyView.text')

    delete other.title
    delete other.text

    return (
      <div
        {...other}
        className={cx('EmptyView', className)}
        >
        {icon &&
          <div className="EmptyView__icon">
            {icon}
          </div>
        }

        {title &&
          <div className="EmptyView__title">
            {title}
          </div>
        }

        {text &&
          <div className="EmptyView__text">
            {text}
          </div>
        }

        {action &&
          <div className="EmptyView__action">
            {action}
          </div>
        }
      </div>
    )
  }
}
