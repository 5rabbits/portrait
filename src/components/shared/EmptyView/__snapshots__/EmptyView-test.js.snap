// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyView should render with minimum props 1`] = `
<EmptyView>
  <div
    className="empty-view"
  >
    <div
      className="empty-container"
    >
      <div
        className="empty-dialog"
      />
    </div>
  </div>
</EmptyView>
`;
